src/index.ts

```diff
@@ -104,27 +108,40 @@ class SpellChecker {
       public app: JupyterFrontEnd, public tracker: INotebookTracker, public palette: ICommandPalette,
       public editor_tracker: IEditorTracker, public status_bar: IStatusBar, public setting_registry: ISettingRegistry
     ){
-        this.language = languages[0];
+        // have at least a default
+        this.language = languages[0]
+
+        // read the settings
+        this.setup_settings();
+
+        // setup the static content of the spellchecker UI
         this.setup_button();
         this.setup_suggestions();
         this.setup_language_picker();
-        this.load_dictionary().catch(console.warn);
+        this.setup_ignore_action()
+
         this.tracker.activeCellChanged.connect(() => this.setup_cell_editor(this.tracker.activeCell));
         // setup newly open editors
         this.editor_tracker.widgetAdded.connect((sender, widget) => this.setup_file_editor(widget.content, true));
         // refresh already open editors when activated (because the MIME type might have changed)
         this.editor_tracker.currentChanged.connect((sender, widget) => this.setup_file_editor(widget.content, false));
-        this.setup_settings()
-        this.setup_ignore_action()
     }
 
+    // move the load_dictionary into the setup routine, because then
+    // we know that the values are set correctly!
     setup_settings() {
         Promise.all([this.setting_registry.load(extension.id), this.app.restored])
           .then(([settings]) => {
               this.update_settings(settings);
               settings.changed.connect(() => {
                   this.update_settings(settings);
+
+                  // load the dictionary
+                  this.load_dictionary().catch(console.warn);
               });
+
+              // load the dictionary
+              this.load_dictionary().catch(console.warn);
```

Comment:
Could `load_dictionary()` be moved into `update_settings()` instead of repeating it twice here?